// bulma-components/lib/Layout/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Footer = exports.Hero = exports.default = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var BulmaContainer = function (_Component) {
  _inherits(BulmaContainer, _Component);

  function BulmaContainer() {
    _classCallCheck(this, BulmaContainer);

    return _possibleConstructorReturn(this, (BulmaContainer.__proto__ || Object.getPrototypeOf(BulmaContainer)).apply(this, arguments));
  }

  _createClass(BulmaContainer, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        { className: 'container ' + this.props.className },
        this.props.children
      );
    }
  }]);

  return BulmaContainer;
}(_react.Component);

var BulmaHero = function (_Component2) {
  _inherits(BulmaHero, _Component2);

  function BulmaHero() {
    _classCallCheck(this, BulmaHero);

    return _possibleConstructorReturn(this, (BulmaHero.__proto__ || Object.getPrototypeOf(BulmaHero)).apply(this, arguments));
  }

  _createClass(BulmaHero, [{
    key: 'injectHeader',
    value: function injectHeader() {
      if (this.props.hasOwnProperty('header')) {
        return _react2.default.createElement(
          'div',
          { className: 'hero-head' },
          this.props.header
        );
      }
    }
  }, {
    key: 'injectFooter',
    value: function injectFooter() {
      if (this.props.hasOwnProperty('footer')) {
        return _react2.default.createElement(
          'div',
          { className: 'hero-foot' },
          this.props.footer
        );
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'section',
        { className: 'hero ' + this.props.className + ' ' + this.props.size },
        this.injectHeader(),
        _react2.default.createElement(
          'div',
          { className: 'hero-body' },
          this.props.children
        ),
        this.injectFooter()
      );
    }
  }]);

  return BulmaHero;
}(_react.Component);

var BulmaFooter = function (_Component3) {
  _inherits(BulmaFooter, _Component3);

  function BulmaFooter() {
    _classCallCheck(this, BulmaFooter);

    return _possibleConstructorReturn(this, (BulmaFooter.__proto__ || Object.getPrototypeOf(BulmaFooter)).apply(this, arguments));
  }

  _createClass(BulmaFooter, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'footer',
        { className: 'footer' },
        this.props.children
      );
    }
  }]);

  return BulmaFooter;
}(_react.Component);

var Hero = BulmaHero;
var Footer = BulmaFooter;

exports.default = BulmaContainer;
exports.Hero = Hero;
exports.Footer = Footer;
